### DISCUSSION TITLE
Rewire,  RewireSystem, RewireLocation, etc

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#184] [General]
I've been messing about with rewire panels and I have a few questions:

I can clone a panel and it seems to work pretty good, but I want to be able to place my own.
The missing panel animation was fixed thanks to modfying the vdu settings in rewirestandalone thanks to help in #181 

I see a few parameters I need help defining for making my own.

display_name
map_keyframe
map_y_offset
map_x_offset
map_zoom

The display name is from the UI script file and I think I got that part.
It's the other four I need help with;  how do I determine those parameters' data?

For the rewire location there is display _name and element_name and shows what looks like same information
as would be used in map_keyframe with an extra number. So in Tech_Hub there's a map_keyframe that says Tech_Hub_1
and it goes to a venthilation(sic) system with an element_name Tech_Hub_16.  Is this an arbitrary name or something 
defined elsewhere.  So for reference this is Tech_Hub but that's just example
And 'world_pos' goes to a position marker.   What is his position supposed to be defined from? The target object 
or just a generic vacinty?  RewireSmoke for example has a position already. Would the world_pos be the same?
Looking and comparing is not easy but it looks like the answer here is 'not quite'.  The world_pos is close to the gascloud
for example in M10 but not the same.  So information on how that is supposed to work is needed.

And lastly, I know it's a work in progress for enums so I hesitate to mention it.  
RewireSystem> display_name_enum and system_type do not automatically set ENUM as their datatype, but 
usually default to string.  Since display_name_enum even has enum in the name it should only allow that type.    

I know that is still a WIP though but I'm mentioning it here since I'm on this subject of rewire.

**Comment by ghramsey** [comment]
Also, cloning rewire panels seems to be acting "fiddly".  

Other night when  I cloned a panel from Lobby into the Stairwell using proxies the panel (once the vdu setting was fixed)
worked exactly as those in the upper lobby do and the map reference points of the affected systems were correctly shown.

Now, I've cloned the rewire panel from the gas cloud in the serverhub and the right  (map) panel is blank.
![image](https://user-images.githubusercontent.com/51859157/232593055-f35f4e3b-36f9-4d8d-a6e2-3f92d7dfccd1.png)

**Comment by ghramsey** [comment]
OH KAY.  I didn't realize that the original location doesn't show a map for that panel either.

So it's a non-issue.   This is the default location:

![image](https://user-images.githubusercontent.com/51859157/232630440-8f7a9a0e-29c3-4833-99f5-88b29cd502b5.png)

**Comment by ghramsey** [comment]
I'm trying to figure out how the rewire_smoke settings work.  I setup a world_pos relative to same area as the smoke, turned  
it on.   No effect.  The element name seems to produce a different location on the map screen of where the affected system is.  
So it's not arbitrary, but is defined somewhere.  Not sure where.

Tech_hub_2_2 is same as the location for one of the rewire_smoke elements in the lobby so it shows the "o2" marker on the catwalk
near where the hostiles spawn in M3 and M10.   I seem to have triggered it to show the door and vent correctly and I can control both
 as I intended.  The rewire_smoke element is not working I created from scratch though is not working.
 
 I might be missing a something I need to setup for the smoke.

**Reply by MattFiler** [reply]
The world position could maybe be relative to the map in 2D space, rather than the environment at 3D space. It's also possible that the map is offset from the world in a way you wouldn't expect (e.g. the tab map has a world origin set to it which offsets all points if I remember correctly). Worth referencing with existing ones to see how the positions line up there.

Apologies for the lack of updates/replies recently - I'm heads down working on some [new asset tools](https://twitter.com/MattFiler/status/1647616122655502339) - more info very shortly :)

**Comment by ghramsey** [comment]
I think you're 100% right.   It's 2D.  And it's relative to the game map screen.  I'm going to have to use a novice game save file.
For some reason this save I've been messing with is NM mode.  Didn't realize that even until I'd tried the map.

No worries on the reply time frame..  That update looks... interesting.
